Web socket reconnect failure #643

mdhsl opened the pull request.

mdhsl committed:
Save the state of extraUrl and queryString to use it with reconnection stuff

mdhsl committed:
Save the state of extraUrl and queryString to use it with reconnection stuff

mdhsl committed:
Add retry limit

nickgaray commented on source/core/datasource/handler/TimeSeriesDataSourceHandler.js:
```diff
@@ -22,12 +22,13 @@ class TimeSeriesDataSourceHandler extends DataSourceHandler {
             if (lastStartTimeCst !== 'now') {
-                this.connector.setUrl(this.parser.buildUrl(
-                    {
+                this.connector.setUrl(endpoint + '?' + this.parser.buildUrl({
```
parser object of this is sometimes undefined, therefore invocation of buildUrl causes exception: 

`TimeSeriesDataSourceHandler.js:35 
        
       Uncaught TypeError: Cannot read properties of undefined (reading 'buildUrl')
    at connector.onReconnect (TimeSeriesDataSourceHandler.js:35:68)
    at WebSocketConnector.createReconnection (WebSocketConnector.js:132:64)
    at ws.onclose (WebSocketConnector.js:108:26)`

nickgaray replied:
The parser object can result in a null reference, this seems like if we get into this situation there is still an error

mdhsl committed:
create getQueryString method to update WS time startTime while reconnecting

mdhsl committed:
remove unused import

mdhsl committed:
- Mofify the architecture to remove parser from building URL into SOS & SweApi
- Fix Sos WS reconnection

mdhsl committed:
- retest all examples
- Fix SosGetFois: missing init(), wrong name (parseData -> parseDataBlock)
- Fix deckLayerView: crash when color was not defined into the layer
- Fix FileConnector/FileDataSource/FileParser: use new way to parse datablock (following SweApi and Sos ones)
- Demos: Fix 3dr-react-tsx demo
